packages/authorization-process/src/services/authorizationService.ts

```diff
@@ -671,6 +675,59 @@ export function authorizationServiceBuilder(
       }
       return key;
     },
+    async getKeyWithClientByKeyId({
+      clientId,
+      kid,
+      logger,
+    }: {
+      clientId: ClientId;
+      kid: string;
+      logger: Logger;
+    }): Promise<KeyWithClient> {
+      logger.info(`Getting client ${clientId} and key ${kid}`);
+      const client = await retrieveClient(clientId, readModelService);
+      const key = client.data.keys.find((key) => key.kid === kid);
+
+      if (!key) {
+        throw keyNotFound(kid, clientId);
+      }
+
+      const pemKey = decodeBase64ToPem(key.encodedPem);
+      const jwk: JsonWebKey = createJWK(pemKey);
+      if (jwk.kty && jwk.kty.toUpperCase() !== "RSA") {
+        throw unknownKeyType(jwk.kty);
+      }
+      const jwkKey: ApiJWKKey = {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        kty: jwk.kty!,
+        keyOps: [],
+        kid: key.kid,
+        x5u: x5u(
+          "Qui dovrebbe andare l'URL del certificato, ma non sò come prenderlo",
+          pemKey
+        ),
+        x5t: crypto.createHash("sha1").update(pemKey).digest("base64"),
+        x5tS256: x5tS256(
+          "Qui dovrebbe andare il certificato DER, ma non sò come prenderlo"
+        ),
+        x5c: x5c([
+          "Qui dovrebbe andare l'array dei certificati DER, ma non sò come prenderlo",
+        ]),
+        d: jwk.d,
+        e: jwk.e,
+        p: jwk.p,
+        q: jwk.q,
+        dp: jwk.dp,
+        dq: jwk.dq,
+        qi: jwk.qi,
+        oth: [], // Non sò dove prendere questo campo
+      };
```

Comment:
```suggestion
      if (jwk.kty?.toUpperCase() !== "RSA") {
        throw unknownKeyType(jwk.kty ?? "unknown");
      }

      const jwkKey: ApiJWKKey = schemas.JWKKey.parse({
        ...jwk,
        kid: "32532535",
      });
```
I think that this should be enough